src/__tests__/index.test.tsx

```diff
@@ -397,6 +398,38 @@ describe('EppoPrecomputedReactNativeClient E2E test', () => {
       format: 'PRECOMPUTED',
     });
   });
+
+  it('deduplicates assignment logging', () => {
+    // Reset the mock logger and assignment cache before this test
+    const assignmentCacheMockLogger = td.object<IAssignmentLogger>();
+    const mockAssignmentCache = td.object<AssignmentCache>();
+    td.when(mockAssignmentCache.has(td.matchers.anything())).thenReturn(
+      false,
+      true
+    );
+    td.when(mockAssignmentCache.set(td.matchers.anything())).thenReturn();
+    globalClient.useCustomAssignmentCache(mockAssignmentCache);
+    globalClient.setAssignmentLogger(assignmentCacheMockLogger);
+
+    expect(
+      td.explain(assignmentCacheMockLogger.logAssignment).callCount
+    ).toEqual(0);
+    globalClient.getStringAssignment('string-flag', 'default');
+    expect(
+      td.explain(assignmentCacheMockLogger.logAssignment).callCount
+    ).toEqual(1);
+    globalClient.getStringAssignment('string-flag', 'default');
+    expect(
+      td.explain(assignmentCacheMockLogger.logAssignment).callCount
+    ).toEqual(1);
+
```

Comment:
or we could have another test case where we could use different subject attributes in another test case to test the flip-flopping case. As long as its tested in one place.

I imagine this will work because you extend `AbstractAssignmentCache` but would protect us should that change

Reply:
Hmm that's a little tough to test because the client is a singleton and doesn't have a `setSubjectAttributes` method since we designed this to be set via the constructor only.

I think a similar direction for a test could be changing the flag key and checking that the logAssignment call increments